src/ast/mod.rs

```diff
@@ -425,6 +425,28 @@ impl fmt::Display for WindowFrameBound {
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub enum Statement {
+    /// Analyze (Hive)
+    Analyze {
+        table_name: ObjectName,
+        partitions: Option<Vec<Expr>>,
+        for_columns: bool,
+        cache_metadata: bool,
+        noscan: bool,
+        compute_statistics: bool,
+    },
+    /// Truncate (Hive)
+    Truncate {
+        table_name: ObjectName,
+        partitions: Option<Vec<Expr>>,
+    },
+    /// Msck (Hive)
+    Msck {
+        table_name: ObjectName,
+        repair: bool,
+        add_partitions: bool,
```

Comment:
Add/drop/sync could be an enum I think? Also simplifies the parser code a bit and allows for easier patter matching.